### PR TITLE
JUCX: use random port and print listener address.

### PR DESCRIPTION
**Pull request by petro-rudenko**
## What
Use random port for JUCX listener test + print address where Listener is bound. 

## Why ?
To better understand fails https://dev.azure.com/ucfconsort/0b36e3f0-8ab9-4a48-b68b-4b2350e02c88/_apis/build/builds/14708/logs/241

### CODE REVIEW

File: bindings/java/src/test/java/org/openucx/jucx/UcpListenerTest.java [hunk 1]
```diff
@@ -75,13 +50,20 @@ static UcpListener tryBindListener(UcpWorker worker, UcpListenerParams params) {
             Collections.list(iface.getInetAddresses()).stream())
             .collect(Collectors.toList());
         for (InetAddress address : addresses) {
-            try {
-                result = worker.newListener(
-                    params.setSockAddr(new InetSocketAddress(address, port)));
-                break;
-            } catch (UcxException ignored) { }
+            for (int i = 0; i < 5; i++) {
```
Comment (yosefe) [inline bindings/java/src/test/java/org/openucx/jucx/UcpListenerTest.java:53]: maybe 10 retries?
to avoid CI restart if fails

**Comment by yosefe** [comment]
error seems relevant
```
2021-03-20T19:05:31.4184607Z [1616267126.840743] [swx-rdmz-ucx-new-01:25165:0]          ucp_ep.c:1168 UCX  WARN  disconnect failed: Endpoint timeout
2021-03-20T19:05:31.4186333Z [1616267131.414007] [swx-rdmz-ucx-new-01:25165:0]            sock.c:406  UCX  ERROR bind(fd=169 addr=fe80::1602:ecff:fe41:2fb4:27256) failed: Address already in use
2021-03-20T19:05:31.4188276Z [1616267131.414116] [swx-rdmz-ucx-new-01:25165:0]            sock.c:406  UCX  ERROR bind(fd=170 addr=10.224.36.118:27256) failed: Address already in use
2021-03-20T19:05:31.4189436Z Bound UcpListner on: /10.224.36.118:27257
2021-03-20T19:05:31.4190799Z [1616267131.417412] [swx-rdmz-ucx-new-01:25165:0]            sock.c:593  UCX  ERROR unknown address family: 0
2021-03-20T19:05:31.4194545Z [1616267131.419164] [swx-rdmz-ucx-new-01:25165:0]       wireup_cm.c:69   UCX  ERROR failed to create a uct sockaddr endpoint on tcp cm 0x7f6f05fb0e70
2021-03-20T19:05:31.4197966Z [swx-rdmz-ucx-new-01:25165:0:25171]   wireup_cm.c:1295 Assertion `uct_cm_ep != NULL' failed
2021-03-20T19:05:31.4415017Z /scrap/azure/agent-01/AZP_WORKSPACE/1/s/src/ucp/wireup/wireup_cm.c: [ ucp_ep_cm_disconnect_cm_lane() ]
2021-03-20T19:05:31.4416526Z       ...
2021-03-20T19:05:31.4417287Z      1292     uct_ep_h uct_cm_ep = ucp_ep_get_cm_uct_ep(ucp_ep);
2021-03-20T19:05:31.4418253Z      1293     ucs_status_t status;
2021-03-20T19:05:31.4418921Z      1294 
2021-03-20T19:05:31.4419643Z ==>  1295     ucs_assert_always(uct_cm_ep != NULL);
2021-03-20T19:05:31.4420418Z      1296     /* No reason to try disconnect twice */
2021-03-20T19:05:31.4421618Z      1297     ucs_assert(!(ucp_ep->flags & UCP_EP_FLAG_DISCONNECTED_CM_LANE));
2021-03-20T19:05:31.4422765Z      1298     ucs_assert(!(ucp_ep->flags & UCP_EP_FLAG_FAILED));
2021-03-20T19:05:31.4628130Z ==== backtrace (tid:  25171) ====
2021-03-20T19:05:31.4630467Z  0 0x00000000000bc679 ucp_ep_cm_disconnect_cm_lane()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/src/ucp/wireup/wireup_cm.c:1295
2021-03-20T19:05:31.4632313Z  1 0x000000000003ef70 ucp_worker_set_ep_failed()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/src/ucp/core/ucp_worker.c:501
2021-03-20T19:05:31.4633956Z  2 0x00000000000bb2fb ucp_cm_client_try_next_cm_progress()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/src/ucp/wireup/wireup_cm.c:71
2021-03-20T19:05:31.4635610Z  3 0x000000000005218b ucs_callbackq_slow_proxy()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/src/ucs/datastruct/callbackq.c:402
2021-03-20T19:05:31.4637254Z  4 0x000000000004511a ucs_callbackq_dispatch()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/src/ucs/datastruct/callbackq.h:211
2021-03-20T19:05:31.4638997Z  5 0x000000000004511a uct_worker_progress()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/src/uct/api/uct.h:2559
2021-03-20T19:05:31.4640827Z  6 0x000000000004511a ucp_worker_progress()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/src/ucp/core/ucp_worker.c:2431
2021-03-20T19:05:31.4641669Z =================================
2021-03-20T19:05:31.4642604Z [swx-rdmz-ucx-new-01:25165:0:25171] Process frozen...
2021-03-20T19:59:46.6109028Z ##[error]The operation was canceled.
```

**Comment by petro-rudenko** [comment]
So here's an issue:
1. It fails to create cm_uct_ep: https://github.com/openucx/ucx/blob/master/src/ucp/wireup/wireup_cm.c#L66
2. `ucp_ep_get_cm_lane(ucp_ep)` is not NULL_LANE, but there's no cm_uct_ep
3. With this commit https://github.com/openucx/ucx/pull/6498/commits/a9abfb8ac7ec511ddfa24a5391d21820dafcd92e - at least it's not faults on assert

**Comment by yosefe** [comment]
@evgeny-leksikov WDYT?

**Comment by petro-rudenko** [comment]
But the root cause is that for some reason in `wireup_ep->cm_remote_sockaddr` is uninitialized. So `ucp_ep_client_cm_create_uct_ep` exits with invalid_param status:
```
 status = ucs_sockaddr_sizeof((struct sockaddr *)&wireup_ep->cm_remote_sockaddr,
                                      &sockaddr_size);
         if (status != UCS_OK) {
             return status;
         }
```

**Comment by evgeny-leksikov** [comment]
> @evgeny-leksikov WDYT?

I think the check added in https://github.com/openucx/ucx/commit/a9abfb8ac7ec511ddfa24a5391d21820dafcd92e should be part of https://github.com/openucx/ucx/blob/55c3e41bd8bb27934759cb2b94a78cc3917a383c/src/ucp/core/ucp_ep.c#L2515

File: src/ucp/core/ucp_ep.c [hunk 2]
```diff
@@ -2487,7 +2487,8 @@ uct_ep_h ucp_ep_get_cm_uct_ep(ucp_ep_h ep)
 
 int ucp_ep_is_cm_local_connected(ucp_ep_h ep)
 {
-    return ucp_ep_has_cm_lane(ep) && (ep->flags & UCP_EP_FLAG_LOCAL_CONNECTED);
+    return ucp_ep_has_cm_lane(ep) && (ep->flags & UCP_EP_FLAG_LOCAL_CONNECTED) &&
```
Comment (yosefe) [inline src/ucp/core/ucp_ep.c:2490]: maybe we can remove ucp_ep_has_cm_lane(ep), rely only on (ucp_ep_get_cm_uct_ep(ep) != NULL)
@evgeny-leksikov WDYT?

Comment (evgeny-leksikov) [reply src/ucp/core/ucp_ep.c:2490]: sure, `ucp_ep_get_cm_uct_ep` already contains `ucp_ep_get_cm_lane() == UCP_NULL_LANE` check

**Review by yosefe — APPROVED** [review]

**Review by evgeny-leksikov — APPROVED** [review]

**Pull request merged**